dessia_common/datatools/dataset.py

```diff
@@ -666,11 +663,20 @@ def _histogram_unic_value(self, idx_col: int, name_attr: str):
     def _tooltip_attributes(self):
         return self.common_attributes
 
-    def _plot_data_list(self):
-        plot_data_list = []
-        for row, _ in enumerate(self.dessia_objects):
-            plot_data_list.append({attr: self.matrix[row][col] for col, attr in enumerate(self.common_attributes)})
-        return plot_data_list
+    def _object_to_sample(self, dessia_object: DessiaObject, reference_path: str, row: int):
+        sample_values = {attr: self.matrix[row][col] for col, attr in enumerate(self.common_attributes)}
+        full_reference_path = f"{reference_path}/dessia_objects/{row}"
+        name = dessia_object.name if dessia_object.name else f"Sample {row}"
+        return sample_values, full_reference_path, name
+
+    def _to_samples(self, reference_path: str):
```

Comment:
reference_path arg could (should ?) be defaulted as "#" here as well. It is also the case in the object_to_sample method. In the case anyone would inherit your class and use these methods, it would allow them to use the feature by default and not break the behavior in their object

Reply:
I was thinking that if plot_data gives a value to reference_path, these methods don't need one but if you tell me the opposite, I put this default value

Reply:
Well, i did it

Reply:
The thing is, someone could use these without going through plot_data